internal/cloud/atlas/clusters.go

```diff
@@ -0,0 +1,44 @@
+package atlas
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+// Cluster contains non sensitive data about an Atlas cluster
+type Cluster struct {
+	ID    string `json:"id"`
+	Name  string `json:"name"`
+	State string `json:"stateName"`
+}
+
+const (
+	clustersByGroupIDPattern = "/api/atlas/v1.0/groups/%s/clusters"
+)
+
+func (c *client) ClustersByGroupID(groupID string) ([]Cluster, error) {
```

Comment:
[nit] I'd name this just `Clusters` (we don't really use "ByGroupID" or "ByAppID" in any of the other routes, even though most require it)

along those same lines, the path pattern can have `ByGroupID` removed as well, and the unexpected status code below can be just "get clusters"